utils/syscall.c

```diff
@@ -382,9 +382,342 @@ static const myst_syscall_pair_t _pairs[] = {
     PAIR(SYS_myst_interrupt_thread),
     PAIR(SYS_myst_pre_launch_hook),
     /* add new entries here! */
+    /* NOTE: when you add a new syscall here, add it to its corresponding group
+       below as well! */
     {0, NULL},
 };
 
+static const myst_syscall_group_t _groups[] = {
```

Comment:
```suggestion
const myst_syscall_group_t _groups[] = {
```
`_groups[]` already has global scope, I don't think you need static here. Most uses of C statics I've seen are for having something like a function level global variable.

Reply:
Also consider moving strace related structures and definitions to a  `strace.h` header file.

Reply:
Makes sense, I removed the static keyword there